src/input.c

```diff
@@ -136,6 +139,16 @@ VbResult input_open_editor(Client *c)
     idreturn = ext_proxy_eval_script_sync(c, "vimb_input_mode_element.id");
     g_variant_get(idreturn, "(bs)", &idSuccess, &id);
 
+    if( !idSuccess || !id || strlen(id) == 0) {
+        element_map_key = element_map_next_key++;
+        char *js_command = g_strdup_printf(
```

Comment:
You could also use JavaScript in `src/scripts/*.js` so this will be minified and escaped so you only need to use the constant instead of the long string. The constant is wirtten to `src/scripts/scripts.h` file.

Reply:
Cool, I'll change that. I only looked at the code inside input.c and did not see you can do that.

Reply:
Should I use this for all javascript within the code or just for the multi line scripts?

Reply:
I think it would be enough to do this for the multiliners or when the code is hard to read because of the heavy quote escaping.
If I'll have some spare time in future I'd like to move all the JavaScript snippets into one or more JavaScript Objects that are injected into the DOM so that we only need to call our little helpers instead of running scripts snippets via dbus on the webextension every time.